### PR TITLE
Update Docker and Go on Jenkins agents, and install jq

### PR DESCRIPTION
**Pull request by ixdy**
Part of fixing #3274. I need to test these changes to make sure they're not going to break anything.

### CODE REVIEW

**Comment by cblecker** [comment]
Updated go, updated docker, *and* jq? As long as it doesn't break anything, this is a winner to me 👍🏻

**Comment by ixdy** [comment]
this appears to be working, which is basically what I expected, since most things run inside docker now.
I could probably even get rid of Go and everything would probably still work.

File: jenkins/agent-ctl.sh [hunk 1]
```diff
@@ -309,7 +309,7 @@ which docker || curl -sSL https://get.docker.com/ | sh
 id jenkins || sudo useradd jenkins -m
 sudo usermod -aG docker jenkins
 
-# Downgrade to 1.9.1
+# Downgrade docker to 1.11.2, the version on COS
 # https://github.com/kubernetes/kubernetes/issues/21451
```
Comment (luxas) [inline jenkins/agent-ctl.sh]: relevant anymore?

Comment (ixdy) [inline jenkins/agent-ctl.sh]: not especially, but it gives at least some history as to why we're pinning to a version instead of using latest.

Comment (luxas) [inline jenkins/agent-ctl.sh]: Yeah, makes sense to keep that then. Feel free to merge!

**Comment by ixdy** [comment]
yep, everything is working, and I'm upgrading the rest of our "heavy" CI agents. I'll tackle light and PR agents sometime later (maybe over the holiday?).

**Pull request merged**